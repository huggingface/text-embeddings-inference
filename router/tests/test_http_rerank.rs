mod common;

use crate::common::{start_server, Score};
use anyhow::Result;
use insta::internals::YamlMatcher;
use serde::{Deserialize, Serialize};
use serde_json::json;
use text_embeddings_backend::DType;

#[derive(Serialize, Deserialize, Debug, PartialEq)]
pub struct SnapshotRank {
    index: usize,
    score: Score,
    text: String,
}

#[tokio::test]
#[cfg(feature = "http")]
async fn test_rerank() -> Result<()> {
    start_server(
        "BAAI/bge-reranker-base".to_string(),
        Some("refs/pr/5".to_string()),
        DType::Float32,
    )
    .await?;

    let request = json!({
        "query": "test",
        "texts": vec!["test", "other", "test"],
        "return_text": true
    });

    let client = reqwest::Client::new();
    let res = client
        .post("http://0.0.0.0:8090/rerank")
        .json(&request)
        .send()
        .await?;

    let ranks = res.json::<Vec<SnapshotRank>>().await?;
    let matcher = YamlMatcher::<Vec<SnapshotRank>>::new();
    insta::assert_yaml_snapshot!("ranks", ranks, &matcher);

    assert_eq!(ranks[0].index, 2);
    assert_eq!(ranks[1].index, 0);
    assert_eq!(ranks[0].score, ranks[1].score);

    Ok(())
}
