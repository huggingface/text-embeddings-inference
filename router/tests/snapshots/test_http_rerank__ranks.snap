---
source: router/tests/test_http_rerank.rs
assertion_line: 42
expression: ranks
---
- index: 2
  score: 0.9997739
  text: test
- index: 0
  score: 0.9997739
  text: test
- index: 1
  score: 0.3638598
  text: other
